mapsforge-map/src/main/java/org/mapsforge/map/layer/cache/InMemoryTileCache.java

```diff
@@ -120,6 +120,10 @@ public synchronized void setCapacity(int capacity) {
 
     @Override
     public synchronized void setWorkingSet(Set<Job> jobs) {
+        int newCapacity = Math.max(jobs.size(), this.lruCache.getCapacity());
+        if (this.lruCache.getCapacity() < newCapacity) {
+            this.lruCache.setCapacity(newCapacity);
+        }
```

Comment:
The InMemoryTileCache should always have at least enough capacity to hold the working set.